docs/modules/ROOT/pages/gsnkit.adoc

```diff
@@ -23,75 +15,97 @@ starter kit you want to use. In this case we'll start our project using the kit
-
-
-``
-openzeppelin init
-``
-
-
-In a new terminal window, in the `client` directory, run the React app:
-
-``
-cd client
-npm run start
-``
-
-## Interact
-
-You can interact directly with your smart contracts from the `openzeppelin` cli.
-
-Commands start with either `openzeppelin` or with the `oz` shorthand.
-[horizontal]
-``openzeppelin transfer``:: To send funds to a given address.
-
-``openzeppelin balance [address]``::  To query the ETH balance of the specified account, also supports ERC20s.
-
-``openzeppelin send-tx``::  To send a transaction to your contract and returns the events.
-
-``openzeppelin call``::  To execute a constant method and receive back the value.
-
-`openzeppelin`:: To see a complete list of available commands.
-
-
-## Test
-
-Truffle can run tests written in Solidity or JavaScript against your smart contracts. Note the command varies slightly if you're in or outside of the truffle development console.
-
-From inside the development console.
-
-``
-test
-``
-
-From outside the development console.
-
-``
-truffle test
-``
-
-Jest is included for testing React components. Compile your contracts before running Jest, or you may receive some file not found errors.
-
-From inside the client directory when running this
-``
-npm run test
-``
-
-## Build
-
-To build the application for production, use the build script. A production build will be in the `client/build` folder.
-
-``
-// ensure you are inside the client directory when running this
-npm run build
-``
+## Why this kit?
+
+This kit leverages GSN to create dapps that are ready for mass adoption. Free your users from
+the initial burden of installing Metamask and obtaining Ether. Create blockchain applications
+that are indistinguishable from Web2.0 apps.
+
+This documents assumes familiarity with Gas Station Network. Here are some resources about it:
+
+https://gsn.openzeppelin.com/[Website]
+https://docs.openzeppelin.com/contracts/2.x/gsn[GSN Contracts Overview]
+https://github.com/OpenZeppelin/openzeppelin-gsn-provider[GSN Provider]
+
+### How does it use Web3 with GSN?
+This kit uses Open Zeppelin https://github.com/OpenZeppelin/openzeppelin-network.js[network.js] to create the connection to Web3. Using a couple
+flags for development and production you can see how the dapp obtains a context that is aware of Gas Station Network.
+
+[source,solidity]
+----
+// get GSN web3
+const context = useWeb3Network('http://127.0.0.1:8545', {
+  gsn: {
+    dev: true,
+  },
+});
+----
+
+### How are the contracts modified to use GSN?
+
+The `Counter` contract is modified to inherit from `RelayRecipient`.
+Also, the counter contract is going to naively pay for all the transactions that are submitted.
+Note how the `acceptRelayedCall` determines this by returning 0.
+
+[source,solidity]
+----
+pragma solidity ^0.5.0;
+
+import "tabookey-gasless/contracts/GsnUtils.sol";
+import "tabookey-gasless/contracts/IRelayHub.sol";
+import "tabookey-gasless/contracts/RelayRecipient.sol";
```

Comment:
Let's use the openzeppelin contracts in the docs instead. We can update this code as soon as we update the contract in the dapp.

Reply:
sounds good